src/main/java/core/basesyntax/figure/Figure.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.common.Drawable;
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.FigureType;
+
+public abstract class Figure implements Drawable {
+    private FigureType type;
+    private Color color;
+
+    public Figure(FigureType type, Color color) {
+        this.type = type;
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    public FigureType getType() {
+        return type;
+    }
+
+    public abstract double getArea();
```

Comment:
There is inconsistency in your solution, some part of behavior defined in interfaces, but another pars is in abstract class, please move methods to interfaces